packages/react-core/src/components/Progress/ProgressContainer.tsx

```diff
@@ -49,14 +54,39 @@ export const ProgressContainer: React.FunctionComponent<ProgressContainerProps>
   parentId,
   label = null,
   variant = null,
-  measureLocation = ProgressMeasureLocation.top
+  measureLocation = ProgressMeasureLocation.top,
+  isTitleTruncated = false,
+  tooltipPosition
 }: ProgressContainerProps) => {
   const StatusIcon = variantToIcon.hasOwnProperty(variant) && variantToIcon[variant];
+  const [tooltip, setTooltip] = React.useState('');
```

Comment:
how about make this a boolean `const [tooltipVisible, setTooltipVisible] = React.useState(false)`
then `onMouseEnter`, set this to true if content is truncated.
Also add `onMouseLeave to set tooltipVisible to false if it's true.
And the tooltip component
```
<Tooltip position={tooltipPosition} content={title} isVisible={tooltipVisible} trigger="manual">
  {Title}
</Tooltip>
```

Reply:
This is less performant (always renders a Tooltip, which previous benchmarks show is expensive) and I have to add the visibility logic not only here, but also in the table transformers and modal box.